### PR TITLE
Match relat data to data dictionary

### PR DESCRIPTION
**Pull request by baarkerlounger**

### CODE REVIEW

File: app/models/validations/household_validations.rb [hunk 1]
```diff
@@ -188,7 +188,7 @@ def validate_person_age_and_gender_match_economic_status(record, person_num)
   end
 
   def validate_partner_count(record)
-    partner_count = (2..8).count { |n| record.public_send("relat#{n}")&.zero? }
+    partner_count = (2..8).count { |n| tenant_is_partner?(record.public_send("relat#{n}")) }
```
Comment (kosiakkatrina) [inline app/models/validations/household_validations.rb]: `record["relat#{n}"]` would that not work?

Comment (baarkerlounger) [reply app/models/validations/household_validations.rb]: Yep it would. Prefer that?

Comment (kosiakkatrina) [reply app/models/validations/household_validations.rb]: Dunno, if it works the same with nil values then maybe?

File: db/schema.rb [hunk 2]
```diff
@@ -224,16 +217,36 @@
...
+    t.boolean "confirmed"
+    t.string "old_id"
+    t.string "old_org_id"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["data_protection_officer_id"], name: "dpo_user_id"
+    t.index ["organisation_id", "data_protection_officer_id", "confirmed"], name: "data_protection_confirmations_unique", unique: true
+    t.index ["organisation_id"], name: "index_data_protection_confirmations_on_organisation_id"
+  end
+
```
Comment (kosiakkatrina) [inline db/schema.rb]: What is this?

Comment (baarkerlounger) [reply db/schema.rb]: :see_no_evil:

**Review by kosiakkatrina — APPROVED** [review]

**Pull request merged**